ledger-core/virtual/integration/constructor_test.go

```diff
@@ -126,65 +131,63 @@ func TestVirtual_Constructor_WithExecutor(t *testing.T) {
-			require.Equal(t, "New", name)
-			require.Equal(t, []byte("arguments"), args)
-		}).
-		Return(nil, []byte("345"), nil)
-
-	mgr := machine.NewManager()
-	err := mgr.RegisterExecutor(machine.Builtin, executorMock)
-	require.NoError(t, err)
-	server.ReplaceMachinesManager(mgr)
-
-	cacheMock := descriptor.NewCacheMock(mc)
-	server.ReplaceCache(cacheMock)
-	cacheMock.ByClassRefMock.Return(
-		descriptor.NewClass(gen.UniqueReference(), gen.UniqueID(), gen.UniqueReference()),
-		descriptor.NewCode(nil, machine.Builtin, gen.UniqueReference()),
-		nil,
+	runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+
+	var (
+		pulseNumber = server.GetPulse().PulseNumber
+		outgoing    = server.RandomLocalWithPulse()
+		objectRef   = reference.NewSelf(outgoing)
 	)
 
-	outgoing := server.RandomLocalWithPulse()
-	objectRef := reference.NewSelf(outgoing)
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
+		Callee:         class,
+		CallSiteMethod: "New",
+		CallOutgoing:   outgoing,
+		Arguments:      []byte("arguments"),
+	}
+	msg := utils.NewRequestWrapper(pulseNumber, &pl).Finalize()
+
+	{
+		requestResult := requestresult.New([]byte("123"), gen.UniqueReference())
+		requestResult.SetActivate(gen.UniqueReference(), class, []byte("234"))
+
+		runnerMock.AddExecutionMock(calculateOutgoing(pl).String()).
+			AddStart(func(execution execution.Context) {
+				require.Equal(t, "New", execution.Request.CallSiteMethod)
+				require.Equal(t, []byte("arguments"), execution.Request.Arguments)
+			}, &executionupdate.ContractExecutionStateUpdate{
+				Type:   executionupdate.Done,
+				Result: requestResult,
+			})
+	}
 
 	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
 	typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
-		require.Equal(t, res.ReturnArguments, []byte("345"))
+		require.Equal(t, res.ReturnArguments, []byte("123"))
 		require.Equal(t, res.Callee, objectRef)
 		require.Equal(t, res.CallOutgoing, outgoing)
 
 		return false // no resend msg
 	})
 
-	msg := makeVStateReportWithState(server.GetPulse().PulseNumber, objectRef, payload.Missing, nil)
-	server.SendMessage(ctx, msg)
-
-	isolation := contract.ConstructorIsolation()
-
-	pl := payload.VCallRequest{
-		CallType:       payload.CTConstructor,
-		CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
-		Callee:         class,
-		CallSiteMethod: "New",
-		CallSequence:   0,
-		CallOutgoing:   outgoing,
-		Arguments:      []byte("arguments"),
+	{
+		msg := makeVStateReportWithState(pulseNumber, objectRef, payload.Missing, nil)
+		server.SendMessage(ctx, msg)
 	}
 
-	msg = utils.NewRequestWrapper(server.GetPulse().PulseNumber, &pl).Finalize()
 	server.SendMessage(ctx, msg)
```

Comment:
I think, it's better to bring this msg initialization closer to use or rename it, because there is `msg` on 187 line too and you've lost me for a second.

Reply:
fixed